core/patina_internal_device_path/src/lib.rs

```diff
@@ -225,8 +228,8 @@ pub fn remaining_device_path(
     let mut b_ptr = b;
     let mut node_count = 0;
     loop {
-        let a_node = unsafe { *a_ptr };
-        let b_node = unsafe { *b_ptr };
+        // SAFETY: Caller must ensure pointers are valid device_paths
```

Comment:
This function should also be unsafe with the caller requirements called out in the function docs.

Reply:
Updated in latest.